File: OCDelegateDdemo/ViewController.m
//
//  ViewController.m
//  OCDelegateDdemo
//
//  Created by 吴楠 on 2018/10/24.
//  Copyright © 2018 wunan. All rights reserved.
//

#import "ViewController.h"
#import "WNSomeModel.h"
#import "WNProtocal.h"
@interface ViewController ()<WNProtocal>
@property (nonatomic,strong)  WNSomeModel *model;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = UIColor.redColor;
    WNSomeModel *model = [WNSomeModel new];
    model.delegate = self;
    self.model = model;
  
}
- (void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
      [self.model test];
}

#pragma mark - 代理方法
- (void)protocalMethod {
    NSLog(@"oc protocalMethod");
}


@end

File: DelegateDemo/ViewController.swift
//
//  ViewController.swift
//  DelegateDemo
//
//  Created by 吴楠 on 2018/10/24.
//  Copyright © 2018 wunan. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    let model = WNSomeModel.init()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.yellow
        model.delegate = self
        
    }
    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
        model.test()
    }
}

extension ViewController:SwiftProtocol{
    func clickAction() {
        print("swift clickAction")
    }
}

